src/etherscan.ts

```diff
@@ -0,0 +1,57 @@
+import axios from "axios";
+import { config } from "./config";
+import { PartyEvent } from "types";
+
+/**
+ * Verify a proxy contract using the etherscan api
+ * etherscan docs - https://docs.etherscan.io/api-endpoints/contracts#verify-proxy-contract
+ * @param {string} address - party address
+ * @returns {boolean} - true if verified, false otherwise
+ */
+export const verifyProxyContract = async (
+  address: string
+): Promise<boolean> => {
+  if (!address) {
+    throw new Error("Missing argument: address");
+  }
+
+  try {
+    const params = new URLSearchParams();
+    params.append("address", address);
+    const resp = await axios.post(
+      `${config.etherscan.apiBase}/?module=contract&action=verifyproxycontract&apikey=${config.etherscan.apiKey}`,
+      params
+    );
+
+    if (!(resp && resp.data)) {
+      throw new Error(`Invalid response from Etherscan API for ${address}`);
+    }
+    const { data } = resp;
+    if (data.status === "1" && data.result) {
+      console.log(
+        `Successfully verified proxy contract for ${address}. guid ${data.result}.`
+      );
+      return true;
+    } else {
+      console.log("data", data);
+      throw new Error(`Could not verify proxy contract for ${address}`);
+    }
+  } catch (error) {
+    console.error(error);
+    return false;
+  }
+};
+
+/**
+ * verify proxy contract on etherscan if the event is a start event
+ * @param {PartyEvent} event
+ * @returns true/false if verified
+ */
+export const verifyProxyContractForStartEvent = async (
```

Comment:
is it worth it to have such a tiny fn? perhaps just combine this with `verifyProxyContract`

Reply:
i split them up so that i could call the function w/ an address when i was testing locally